uk.co.compucorp.civicrm.appraisals/CRM/Appraisals/BAO/AppraisalCycle.php

```diff
@@ -240,33 +241,106 @@ public static function getStatusOverview($managerId) {
-                (a_overdue.status_id = 3 AND a_overdue.grade_due < NOW())
-            ) AND
-                a_overdue.appraisal_cycle_id = a.appraisal_cycle_id AND
-                a_overdue.status_id = a.status_id
-            )
-            AS overdue FROM civicrm_appraisal a
-            INNER JOIN civicrm_appraisal_cycle ac ON ac.id = a.appraisal_cycle_id
-            WHERE ac.cycle_is_active = 1
+                ac.cycle_is_active = 1
+                %cycles_ids
+                %period_criteria
             GROUP BY a.status_id, a.appraisal_cycle_id
             ORDER BY a.status_id ASC
         ) r
         GROUP BY status_id';
+
         $params = array();
+
+        self::fillQueryPlaceholder($query, $params, array(
+            'name' => 'cycles_ids',
+            'type' => 'Text',
+            'string' => 'AND ac.id IN (%paramIndex_1)',
+            'values' => array($cyclesIds)
+        ));
+        self::fillQueryPlaceholder($query, $params, array(
+            'name' => 'overdue_criteria',
+            'type' => $currentDate === 'NOW()' ? 'Text' : 'String',
+            'string' => self::dueDatesClause('a_overdue', $currentDate, '< %paramIndex_1'),
+            'values' => array($currentDate)
+        ));
+        self::fillQueryPlaceholder($query, $params, array(
+            'name' => 'period_criteria',
+            'type' => 'String',
+            'string' => self::dueDatesClause('a', $startDate, 'BETWEEN %paramIndex_1 AND %paramIndex_2'),
+            'values' => array($startDate, $endDate)
+        ));
+
         $result = CRM_Core_DAO::executeQuery($query, $params);
+
         while ($result->fetch()) {
             $data[$result->status_id]['contacts_count']['due'] = (int)$result->total - (int)$result->overdue;
             $data[$result->status_id]['contacts_count']['overdue'] = (int)$result->overdue;
         }
+
         return $data;
     }
-    
+
+    /**
+     * Fills a placeholder in the given query string using the options passed
+     *
+     * @param {string} $query - The query containing the placeholder
+     * @param {Array} $params - The array used by CRM_Core_DAO::executeQuery to
+     *     interpolate the values in the query before executing it
+     * @param {Array} $options - An array with following properties:
+     *     name - The name of the placeholder that needs to be replaced
+     *     string - The string that replaces the placeholder. The string must
+     *       contain a placeholder %paramIndex_<index> (i.e.: $paramIndex_1)
+     *       that will be used by CRM_Core_DAO::executeQuery to fill in the real value
+     *     values - An array of value that CRM_Core_DAO::executeQuery will
+     *       use in the final query
+     *     type - The type of the values used by CRM_Core_DAO::executeQuery
+     */
+    private function fillQueryPlaceholder(&$query, &$params, $options) {
+        $replacement = '';
+
+        if (!empty(array_filter($options['values']))) {
```

Comment:
You use array_filter here to remove 0s, empty strings, nulls, etc. from $option['values'] and check if that's not empty. But later on you just iterate over the whole array.